Source/WebCore/html/HTMLFormControlElement.cpp

```diff
@@ -308,6 +325,7 @@ void HTMLFormControlElement::dispatchFormControlChangeEvent()
 {
     dispatchChangeEvent();
     setChangedSinceLastFormControlChangeEvent(false);
+    setInteractedSinceLastFormSubmitEvent(true);
```

Comment:
Can dispatchFormControlChangeEvent() be called for script-driven changes? Is it OK to update valid/invalid in that cause?

Reply:
> Can dispatchFormControlChangeEvent() be called for script-driven changes? 

I just audited every call of `dispatchFormControlChangeEvent` and it does only seem to be called in response of user interaction.